src/main/java/be/ugent/rml/Executor.java

```diff
@@ -237,6 +259,67 @@ private List<PredicateObjectGraph> generatePredicateObjectGraphs(Mapping mapping
         return results;
     }
 
+    private Collection<ProvenancedTerm> getIRIsWithSimpleJoin(Record record, Term parentTriplesMap, Map<String, Object[]> parameters) throws Exception {
+        Mapping mapping = this.mappings.get(parentTriplesMap);
+
+        Extractor parentExtractor = extract(parameters, Vocabulary.PARENT_PARAMETER);
+        Extractor childExtractor = extract(parameters, Vocabulary.CHILD_PARAMETER);
+
+        //put in cache
+        //real index instead of 0
+        MultiValuedMap<List<Object>, ProvenancedTerm> recordMap = this.getCachedRecordMap(Pair.of(parentTriplesMap, parentExtractor));
+        if(recordMap == null) {
+            List<Record> records = this.getRecords(parentTriplesMap);
+            recordMap = IntStream.range(0, records.size())
+                    .boxed()
+                    .collect(ArrayListValuedHashMap::new, (m, i) -> {
+                        try {
+                            Record r = records.get(i);
+                            m.put(parentExtractor.extract(r), this.getSubject(parentTriplesMap, mapping, r, i));
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }, AbstractListValuedMap::putAll);
+
+            this.cachedRecordMap.put(Pair.of(parentTriplesMap, parentExtractor), recordMap);
+        }
+
+        return recordMap.get(childExtractor.extract(record));
+    }
+
+    private MultiValuedMap<List<Object>, ProvenancedTerm> getCachedRecordMap(Pair<Term, Extractor> key) {
+        return this.cachedRecordMap.get(key);
+    }
+
+    private Extractor extract(Map<String, Object[]> parameters, String parameterName) {
+        return parameters.values().stream()
+                .filter(a -> a.length == 2)
+                .filter(a -> parameterName.equals(a[0]))
+                .map(a -> a[1])
+                .filter(Extractor.class::isInstance)
+                .map(Extractor.class::cast)
+                .findAny()
+                .orElse(null);
+    }
+
+    // Check if the 2 parameters exist
+    private boolean isSimpleJoin(PredicateObjectGraphMapping pogMapping, Map<String, Object[]> parameters) {
```

Comment:
This function must be documented, what is a 'simple join'?
What are the exact criteria to be a 'simple join' or a normal one?